api/app/signals/apps/email_integrations/tasks.py

```diff
@@ -1,9 +1,70 @@
 # SPDX-License-Identifier: MPL-2.0
 # Copyright (C) 2021 - 2022 Gemeente Amsterdam
+import logging
+
+from django.contrib.auth import get_user_model
+
 from signals.apps.email_integrations.services import MailService
+from signals.apps.signals.models import Department, Signal
 from signals.celery import app
 
+logger = logging.getLogger(__name__)
+User = get_user_model()
+
 
 @app.task
 def send_mail_reporter(pk):
     MailService.status_mail(signal=pk)
+
+
+@app.task
+def send_mail_assigned_signal_departments(signal_pk, department_pks):
+    try:
+        signal = Signal.objects.get(pk=signal_pk)
+    except Signal.DoesNotExist:
+        logger.exception()
+        return
+
+    for department_pk in department_pks:
```

Comment:
This should be part of the `AssignedAction`.

Reply:
Small remark as an interested observer.

That is quite a change, the behavior of the `AbstractSystemMail` changes from sending max one email to potentially many. Moving a check equivalent to the query below to a rule could mean you do a query per user profile instead of filtering out the ones you want beforehand. In stead you could get the relevant users in the `get_recipient_list` method, but then the `send_mail` method must be changed to make sure not all recipients see each other's email addresses.

Interesting and I am curious to see what implementation we settle for. I will add that my PR#261 also uses the `get_recipient_list` method to be able to modify the recipient of a status change email.

Reply:
@vanbuiten and I discussed this yesterday and decided to keep the business logic in tasks.py for now and use the different usages of MailService later as an inspiration for refactoring.